src/Raven.Server/Documents/Handlers/Processors/Queries/AbstractQueriesHandlerProcessorForGet.cs

```diff
@@ -119,77 +121,9 @@ public override async ValueTask ExecuteAsync()
-                    {
-                        await HandleSuggestQueryAsync(indexQuery, queryContext, context, existingResultEtag, token);
-                        return;
-                    }
-
-                    QueryResultServerSide<TQueryResult> result = null;
-                    try
-                    {
-                        result = await GetQueryResultsAsync(indexQuery, queryContext, existingResultEtag, parameters.MetadataOnly, token).AsTask();
-                    }
-                    catch (IndexDoesNotExistException)
-                    {
-                        result?.Dispose();
-                        HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
-                        return;
-                    }
-                    catch (Exception)
-                    {
-                        result?.Dispose();
-                        throw;
-                    }
-
-                    using (result)
-                    {
-                        if (result.NotModified)
-                        {
-                            HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
-                            return;
-                        }
-
-                        HttpContext.Response.Headers[Constants.Headers.Etag] = CharExtensions.ToInvariantString(result.ResultEtag);
-
-                        long numberOfResults;
-                        long totalDocumentsSizeInBytes;
-                        await using (var writer = new AsyncBlittableJsonTextWriter(context, RequestHandler.ResponseBodyStream(), token.Token))
-                        {
-                            result.Timings = indexQuery.Timings?.ToTimings();
-
-                            (numberOfResults, totalDocumentsSizeInBytes) = await writer.WriteDocumentQueryResultAsync(context, result, parameters.MetadataOnly,
-                                WriteAdditionalData(indexQuery, parameters.IncludeServerSideQuery), token.Token);
-                            await writer.MaybeFlushAsync(token.Token);
-                        }
-
-
-                        QueryMetadataCache.MaybeAddToCache(indexQuery.Metadata, result.IndexName);
-
-                        if (RequestHandler.ShouldAddPagingPerformanceHint(numberOfResults))
-                        {
-                            RequestHandler.AddPagingPerformanceHint(PagingOperationType.Queries, $"Query ({result.IndexName})",
-                                $"{indexQuery.Metadata.QueryText}\n{indexQuery.QueryParameters}", numberOfResults, indexQuery.PageSize, result.DurationInMs,
-                                totalDocumentsSizeInBytes);
-                        }
-
-                        AddQueryTimingsToTrafficWatch(indexQuery);
-                    }
+                    var processQueryTask = ProcessQueryAsync(parameters, indexQuery, queryContext, context, existingResultEtag, token);
+                    if (processQueryTask.IsCompletedSuccessfully == false)
```

Comment:
Similar to the above. I assume this will be generated automatically by the compiler.